services-directions/src/main/java/com/mapbox/api/directions/v5/models/BannerView.java

```diff
@@ -0,0 +1,168 @@
+
+  /**
+   * This indicates the mode of the maneuver. If type is of turn, the modifier indicates the
+   * change in direction accomplished through the turn. If the type is of depart/arrive, the
+   * modifier indicates the position of waypoint from the current direction of travel.
+   *
+   * @return String with modifier
+   * @since 5.0.0
+   */
+  @Nullable
+  public abstract String modifier();
+
+  /**
+   * Convert the current {@link BannerView} to its builder holding the currently assigned
+   * values. This allows you to modify a single property and then rebuild the object resulting in
+   * an updated and modified {@link BannerView}.
+   *
+   * @return a {@link BannerView.Builder} with the same values set to match the ones defined
+   *   in this {@link BannerView}
+   * @since 5.0.0
+   */
+  public abstract BannerView.Builder toBuilder();
+
+  /**
+   * Gson type adapter for parsing Gson to this class.
+   *
+   * @param gson the built {@link Gson} object
+   * @return the type adapter for this class
+   * @since 5.0.0
+   */
+  public static TypeAdapter<BannerView> typeAdapter(Gson gson) {
+    return new AutoValue_BannerView.GsonTypeAdapter(gson);
+  }
+
+  /**
+   * Create a new instance of this class by passing in a formatted valid JSON String.
+   *
+   * @param json a formatted valid JSON string defining a BannerText
+   * @return a new instance of this class defined by the values passed inside this static factory
+   *   method
+   * @since 5.0.0
+   */
+  public static BannerView fromJson(String json) {
+    GsonBuilder gson = new GsonBuilder();
+    gson.registerTypeAdapterFactory(DirectionsAdapterFactory.create());
+    return gson.create().fromJson(json, BannerView.class);
+  }
+
+  /**
+   * This builder can be used to set the values describing the {@link BannerView}.
+   *
+   * @since 5.0.0
+   */
+  @AutoValue.Builder
+  public abstract static class Builder {
+
+    /**
+     * Plain text with all the {@link BannerComponents} text combined.
+     *
+     * @param text plain text with all the {@link BannerComponents} text items combined
+     * @return this builder for chaining options together
+     * @since 5.0.0
+     */
+    public abstract Builder text(@NonNull String text);
+
+    /**
+     * A part or element of the {@link BannerInstructions}.
+     *
+     * @param components a {@link BannerComponents} specific to a {@link LegStep}
+     * @return this builder for chaining options together
+     * @since 5.0.0
+     */
+    public abstract Builder components(List<BannerComponents> components);
+
+    /**
+     * This indicates the type of maneuver. See {@link BannerView#type()} for a full list of
+     * options.
+     *
+     * @param type String with type of maneuver
+     * @return this builder for chaining options together
+     * @see StepManeuver.StepManeuverType
+     * @since 5.0.0
+     */
+    public abstract Builder type(@Nullable @StepManeuver.StepManeuverType  String type);
```

Comment:
Based on the OP comment in https://github.com/mapbox/bumblebee/pull/532, it seems that a `StepManeuverType` could be `guidance-view`. Do we need to add `guidance-view` as a new `public static final` in https://github.com/mapbox/mapbox-java/blob/master/services-directions/src/main/java/com/mapbox/api/directions/v5/models/StepManeuver.java ?

Reply:
```
"view": {
  "text": "",
  "components": [
    {
      "text": "",
      "type": "guidance-view",
      "imageURL": "https://api-turn-here-staging-451578336.us-east-1.elb.amazonaws.com/guidance-views/v1/z/jct/CA075101?arrow_ids=CA07510E"
    }
  ],
  "type": "off ramp",
  "modifier": "right"
}
```

The type is within the component, so it's not a new `StepManeuverType`. The component type doesn't have this
https://github.com/mapbox/mapbox-java/blob/5d81814bd9b87c53b9c116520b679af876c5d2ce/services-directions/src/main/java/com/mapbox/api/directions/v5/models/BannerComponents.java#L67

Reply:
But seems like something we could add. @danpaz do you have a list of the different banner component types?

Reply:
@kmadsen yes, component types are documented [here](https://docs.mapbox.com/api/navigation/#banner-instruction-object).

Reply:
copied the documentation into a type https://github.com/mapbox/mapbox-java/pull/1098